### ISSUE TITLE
Add Package object to Product

### ISSUE THREAD

**Issue by zschmerber** [ocsf/ocsf-schema#1053]
I would like to add the `Package` object to the `Product` object and an optional object to use.
Product.Package
<img width="1671" alt="image" src="https://github.com/ocsf/ocsf-schema/assets/13574947/1ab76a98-c525-4b4a-80b2-974d9b4eb4b1">

We have logs that contain application architecture and no way to capture it in the Product Object.

**Comment by pagbabian-splunk** [comment]
Agreed.  Also, `hw_device_info` does not have an `architecture` attribute and probably should.  Where `metadata.product` is what the event source is, other uses of Product can be different, e.g. Application Lifecycle's `app` attribute is of type `Product` and any Discovery of products event (don't think we actually have that, just devices, users, configs).

**Comment by zschmerber** [comment]
pr #1278 fixed this and it can now be closed
